packages/flutter/test/material/range_slider_test.dart

```diff
@@ -833,6 +833,123 @@ void main() {
+              child: Material(
+                child: Center(
+                  child: RangeSlider(
+                    values: values,
+                    min: 0,
+                    max: 100,
+                    onChanged: (RangeValues newValues) {
+                      setState(() {
+                        values = newValues;
+                      });
+                    },
+                    onChangeStart: (RangeValues newValues) {
+                      startValues = newValues;
+                    },
+                    onChangeEnd: (RangeValues newValues) {
+                      endValues = newValues;
+                    },
+                  ),
+                ),
+              ),
+            );
+          },
+        ),
+      ),
+    );
+
+    // Get the bounds of the track by shifting by the the overlay radius.
+    final Offset topLeft = tester.getTopLeft(find.byType(RangeSlider)).translate(24, 0);
+    final Offset bottomRight = tester.getBottomRight(find.byType(RangeSlider)).translate(-24, 0);
+
+    // Drag the start thumb towards the center.
+    final Offset leftTarget = topLeft + (bottomRight - topLeft) * 0.3;
+    expect(startValues, null);
+    expect(endValues, null);
+    await tester.dragFrom(leftTarget, (bottomRight - topLeft) * 0.2);
+    expect(startValues.start, closeTo(30, 1));
+    expect(startValues.end, closeTo(70, 1));
+    expect(values.start, closeTo(50, 1));
+    expect(values.end, closeTo(70, 1));
+    expect(endValues.start, closeTo(50, 1));
+    expect(endValues.end, closeTo(70, 1));
+  });
+
+  testWidgets('Range Slider onChangeEnd and onChangeStart are called on an interaction initiated by drag', (WidgetTester tester) async {
+    RangeValues values = const RangeValues(30, 70);
+    RangeValues startValues;
+    RangeValues endValues;
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return MediaQuery(
+              data: MediaQueryData.fromWindow(window),
+              child: Material(
+                child: Center(
+                  child: RangeSlider(
+                    values: values,
+                    min: 0,
+                    max: 100,
+                    onChanged: (RangeValues newValues) {
+                      setState(() {
+                        values = newValues;
+                      });
+                    },
+                    onChangeStart: (RangeValues newValues) {
+                      startValues = newValues;
+                    },
+                    onChangeEnd: (RangeValues newValues) {
+                      endValues = newValues;
+                    },
+                  ),
+                ),
+              ),
+            );
+          },
+        ),
+      ),
+    );
+
+    // Get the bounds of the track by shifting by the the overlay radius.
```

Comment:
This comment "shifting by the..." doesn't seem to match the code.

Reply:
Updated here and everywhere else